arch/x86/core/acpi.c

```diff
@@ -137,7 +137,7 @@ void *z_acpi_find_table(uint32_t signature)
 		return NULL;
 	}
 
-	if (rsdp->rsdt_ptr) {
+	if (rsdp->rsdt_ptr != 0U) {
```

Comment:
Why the compare to 0 instead of a compare to NULL like you do in printf.h changes?

Reply:
The type of rsdt_ptr field is uint32_t.